### ISSUE TITLE
Support CHECK

### ISSUE THREAD

**Issue by tkellen** [knex/knex#266]
@tgriesser, is there currently a way to do CHECK(fieldName <> '') with the schema builder?

**Comment by roackb2** [comment]
May I know is `check` supported now and is there any documentation or example?
I'm eager to use check constraints with PostgreSQL.

**Comment by elhigu** [comment]
You need to use raw query for that

**Comment by cour64** [comment]
Just dropping in to say, as of v8.0.16 MySQL now fully [supports](https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-16.html) the CHECK constraint.

**Comment by elhigu** [comment]
In that case support for them are welcome :)

**Comment by mdoliner-st** [comment]
Just want to reference https://github.com/knex/knex/pull/2364 so this issue gets linked to it. It's a work-in-progress PR (and maybe very out of date/abandoned) to add a `.check()` function for schema building.

**Comment by tkellen** [comment]
What a blast from the past.

For those still living in a world where they think a javascript DSL in front of their database migrations is desirable, I implore to you to reconsider your approach. This whole concept is ridiculous and unnecessary.

It's quite easy to use any functionality your database engine provides when you use plain SQL migrations. You may also find this useful: https://sqitch.org/

Good luck out there ya'll--I'm unsubscribing from this issue now.

**Comment by sebastian-nowak** [comment]
Yikes, this issue is almost 7 years old and it's still not supported. Is there a way to chain `.raw()` with `table.integer()` to add a `check` that I'm missing?

**Comment by kibertoad** [comment]
@sebastian-nowak PR implementing this is welcome!

**Comment by sebastorama** [comment]
I don't really mind messing with raw migrations. One problem that I'm having is that I can't handle errors from PG check constraints. As in:

```javascript
   describe('name', () => {
    it('should not have special chars', async () => {
      expect.assertions(1);

      try {
        await db('organizations').insert({ name: 'abscde$' });
      } catch (e) {}
      // fake assertion just to prove that the exception is not handled in any form
      expect(2 + 2).toEqual(4);
    });
  });
```
I have a check constraint not allowing special chars for the name field, yet this test fails:
```
 FAIL  src/db/test/organizations.spec.ts
  organizations table
    name
      ✕ is required to be present (127 ms)

  ● organizations table › name › is required to be present

    error: insert into "organizations" ("name") values ($1) returning * - new row for relation "organizations" violates check constraint "organizations_name_regex_ck"

      at Parser.parseErrorMessage (../node_modules/pg-protocol/src/parser.ts:357:11)
      at Parser.handlePacket (../node_modules/pg-protocol/src/parser.ts:186:21)
      at Parser.parse (../node_modules/pg-protocol/src/parser.ts:101:30)
      at Socket.<anonymous> (../node_modules/pg-protocol/src/index.ts:7:48)

  ● organizations table › name › is required to be present

    expect.assertions(1)

    Expected one assertion to be called but received zero assertion calls.

      14 |   describe('name', () => {
      15 |     it('is required to be present', async () => {
    > 16 |       expect.assertions(1);
         |              ^
      17 |
      18 |       try {
      19 |         return await db('organizations').insert({ name: 'abscde$' });

      at Object.<anonymous> (db/test/organizations.spec.ts:16:14)
```
It's possible to do a migration with just raw sql, no big deal, you tend to deal with migrations only a few times, but I have no idea how to handle this error. Or even test for it.

**Comment by OlivierCavadenti** [comment]
> Yikes, this issue is almost 7 years old and it's still not supported. Is there a way to chain `.raw()` with `table.integer()` to add a `check` that I'm missing?

I finish a long work on Json support in Knex and then I plan to finish this PR https://github.com/knex/knex/pull/2364. Be patient some additional weeks.

**Comment by kibertoad** [comment]
Released in 1.0.1.
